src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -218,15 +218,66 @@ internal sealed class EvaluationContext : EvaluationContextBase
                 debugInfo);
         }
 
-        internal CompilationContext CreateCompilationContext(CSharpSyntaxNode syntax)
+        internal CompilationContext CreateCompilationContext()
         {
             return new CompilationContext(
                 this.Compilation,
                 _currentFrame,
                 _locals,
                 _inScopeHoistedLocals,
-                _methodDebugInfo,
-                syntax);
+                _methodDebugInfo);
+        }
+
+        internal byte[] CompileExpressions(
+            ImmutableArray<string> expressions,
+            out ImmutableArray<int> methodTokens,
+            out ImmutableArray<string> errorMessages)
+        {
+            var diagnostics = DiagnosticBag.GetInstance();
+            var syntaxNodes = expressions.SelectAsArray(expr => Parse(expr, treatAsExpression: true, diagnostics: diagnostics, formatSpecifiers: out var formatSpecifiers));
+            byte[] assembly = null;
+            if (!diagnostics.HasAnyErrors())
+            {
+                Debug.Assert(syntaxNodes.All(s => s != null));
+                var context = this.CreateCompilationContext();
+                var moduleBuilder = context.CompileExpressions(syntaxNodes, TypeName, MethodName, diagnostics);
+                if (moduleBuilder != null)
+                {
+                    using (var stream = new MemoryStream())
+                    {
+                        Cci.PeWriter.WritePeToStream(
+                            new EmitContext(moduleBuilder, null, diagnostics),
+                            context.MessageProvider,
+                            () => stream,
+                            getPortablePdbStreamOpt: null,
+                            nativePdbWriterOpt: null,
+                            pdbPathOpt: null,
+                            allowMissingMethodBodies: false,
+                            isDeterministic: false,
+                            cancellationToken: default(CancellationToken));
+                        if (!diagnostics.HasAnyErrors())
+                        {
+                            assembly = stream.ToArray();
+                        }
+                    }
+                }
+            }
+            if (assembly == null)
+            {
+                methodTokens = ImmutableArray<int>.Empty;
+                errorMessages = ImmutableArray.CreateRange(
+                    diagnostics.AsEnumerable().
+                        Where(d => d.Severity == DiagnosticSeverity.Error).
+                        Select(d => GetErrorMessage(d, CSharpDiagnosticFormatter.Instance, preferredUICulture: null)));
```

Comment:
Did we ever make the general change to drop the line/col information from the error messages?  I thought we did, and I was just looking at the test baselines and thinking it's probably superfluous.